Bottom: 18647d425b0f3062b4b3c7ddeec622c45da8a476
Top:    2b1cc21f3744303fdbe2e20c46bb49069dc36d83
Author: Parth Shah <parth@linux.ibm.com>
Date:   2020-01-09 19:02:16 +0530

Refresh of sched-introduce-switch-to

---

diff --git a/kernel/sched/core.c b/kernel/sched/core.c
index 2d360796859d..47c76b5daf83 100644
--- a/kernel/sched/core.c
+++ b/kernel/sched/core.c
@@ -73,6 +73,7 @@ __read_mostly int scheduler_running;
  */
 int sysctl_sched_rt_runtime = 950000;
 
+#ifdef CONFIG_SCHED_SMT
 DEFINE_STATIC_KEY_FALSE(__turbo_sched_enabled);
 static DEFINE_MUTEX(turbo_sched_lock);
 static int turbo_sched_count;
@@ -92,6 +93,10 @@ void turbo_sched_put(void)
 		static_branch_disable(&__turbo_sched_enabled);
 	mutex_unlock(&turbo_sched_lock);
 }
+#else
+void turbo_sched_get(void) { return ; }
+void turbo_sched_get(void) { return ; }
+#endif
 
 /*
  * __task_rq_lock - lock the rq @p resides on.
diff --git a/kernel/sched/sched.h b/kernel/sched/sched.h
index 8d7bb94ae5fd..8c33c56dbae2 100644
--- a/kernel/sched/sched.h
+++ b/kernel/sched/sched.h
@@ -2504,9 +2504,12 @@ static inline void membarrier_switch_mm(struct rq *rq,
 
 void turbo_sched_get(void);
 void turbo_sched_put(void);
+
+#ifdef CONFIG_SCHED_SMT
 DECLARE_STATIC_KEY_FALSE(__turbo_sched_enabled);
 
 static inline bool is_turbosched_enabled(void)
 {
 	return static_branch_unlikely(&__turbo_sched_enabled);
 }
+#endif
